pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -84,7 +84,12 @@ private void setMergedValue(GenericRow row, String column, @Nullable Object merg
       row.putValue(column, mergedValue);
     } else {
       // if column exists but mapped to a null value then merger result was a null value
-      row.putDefaultNullValue(column, _fieldSpecMap.get(column).getDefaultNullValue());
+      // check if column is a multivalued column
+      if (_fieldSpecMap.get(column).isSingleValueField()) {
+        row.putDefaultNullValue(column, _fieldSpecMap.get(column).getDefaultNullValue());
```

Comment:
Should we put null in case of singleValueField as well to keep it consistent with normal full-upsert/partial-upsert merger logic?

Reply:
For MV column, you'll want to put `new Object[]{defaultNullValue}`. Take a look at the logic in `NullValueTransformer`

Reply:
Thanks @Jackie-Jiang  for pointing this. I was concerned about creating new Object[] every time, this approach is cleaner. Will update the PR accordingly.